src/diffusers/schedulers/scheduling_pndm.py

```diff
@@ -270,7 +270,8 @@ def add_noise(
         noise: Union[torch.FloatTensor, np.ndarray],
         timesteps: Union[torch.IntTensor, np.ndarray],
     ) -> torch.Tensor:
-
+        # mps requires indices to be in the same device, so we use cpu as is the default with cuda
+        timesteps = timesteps.to(self.alphas_cumprod.device)
```

Comment:
do we need this in other schedulers as well for ex DDIM, K-LMS ?

Reply:
Not sure. Scheduler tests passed, but they might not be comprehensive enough. I tested LMS by configuring the pipeline manually, will run the others too.

Reply:
I verified them and updated `ScoreSdeVeScheduler`, the others do not use the same type of indexing and it works.